core/src/main/java/org/apache/accumulo/core/client/admin/NamespaceOperations.java

```diff
@@ -202,11 +202,32 @@ void removeProperty(String namespace, String property)
    *           if the user does not have permission
    * @throws NamespaceNotFoundException
    *           if the specified namespace doesn't exist
-   * @since 1.6.0
+   * @deprecated since 2.1.0; use {@link #getPropertiesMap(String)} instead.
    */
+  @Deprecated(since = "2.1.0")
```

Comment:
I am wondering if we _need_ to mark this method deprecated. I can't think of a risky/harmful reason why a user shouldn't use this method. Yes, a map is more convenient, which makes the new method nice but I don't know if we need to mark this as deprecated. If a user is using it today there is no reason why they should change there code unless they want to.

Reply:
I think the initial reasoning that was detailed in the JIRA ticket was bloating the API. In this case, it isn't much bloat but that was the reasoning given in that discussion.

Reply:
I marked it on the PR since I think it warranted discussion. But after some more thought, I think it should be deprecated as well. It does feel like a mistake that the API didn't originally return the more flexible object.

Reply:
My preference would be to not deprecate or remove the method for the following reasons.

- The method is useful for the case of looping through the properties to find props matching a pattern.  This use case is not uncommon and it seems like the existing method supports slightly shorter code.
- For table props, it seems like the existing method has been the only way to get table props in the API since Accumulo was released. So for the past decade any code written would have had to use this method.  To me, removing the method when it has no problems seems needlessly disruptive to this existing code.

My preference would be a `@see` javadoc tag pointing to the other method instead of deprecating.

Reply:
> It does feel like a mistake that the API didn't originally return the more flexible object.

Following that line of thought, maybe Map is not flexible enough either.  For example [a specific configuration type was created for plugins](https://github.com/apache/accumulo/blob/8a636a3ba91f5dae1d8b09b095178889a7d79c1d/core/src/main/java/org/apache/accumulo/core/client/PluginEnvironment.java#L41) and it supports an operation like `isSet()`.

Reply:
I think we'd have to add a lot more plumbing changes to make something like `isSet` or anything else in a dedicated type useful. Personally, I think Map is more than sufficient for now.

Reply:
Since there are two use cases for each type, it might be better to have both methods. Also, the method that returns the iterable is used quite a lot across the Accumulo code (a quick search showed me 51 places). I guess today I am back to not deprecating the method. If the implementation uses the same underlying method and without something better (like a new type that supports `isSet()`) it doesn't seem necessary to deprecate. Sorry, API design is difficult!

Reply:
I'm still in favor of deprecating. Some of the objections so far have been against removing, not against deprecating. The bar for deprecating should be much lower. I think API bloat is a harm that should be considered, and I think the SemVer process mitigates against the concerns over the worst effects of disruption. I would like to clean up API bloat over time, to streamline and simplify our API for ease of use in future, as well as accommodate existing users. I think the deprecation process with SemVer helps balance those competing goals, and think that some of the objections would have our API bloat over time. But, I think API evolution should also include the loss of vestigial APIs.

However, as it stands, I'm outvoted here. So, unless others agree with me, that we should slowly phase out the old vestigial API, using deprecation as a start, we should keep both and not deprecate at this time.

Reply:
My thoughts on deprecating are not super well defined, but my thinking on deprecation of API is that it should be related to functionality that will be removed or is fundamentally broken.  For this particular case the functionality behind the existing API in questions works fine and will not be removed.  Also the new API being introduced is functionally equivalent to the existing API (each could be fully implemented using only the other).  We could introduce an infinite number of functionally equivalent APIs and deprecate existing APIs.

Unrelated to deprecation, I also tend to think that new APIs should normally be motivated by new functionality.  However its not that well defined for me, because I can think of exceptions to this guide.   In this case there is no new functionality, but the new API sill makes sense to me for other reasons.  Thinking about this a bit more, it makes me wonder if there are any new functional changes related to configuration on the horizon.  If so could those require new APIs?  If that were the case it may be prudent to wait on this change and consider it w/ that new functionality.